gridstatus/caiso.py

```diff
@@ -1168,13 +1158,24 @@ def _process_solar_and_wind_forecast_dam(
             values="MW",
         ).reset_index()
 
-        df = self._add_forecast_publish_time(
-            df,
-            current_time,
-            # Day-ahead hourly wind and solar forecast is published at 7:00 AM according
-            # to OASIS.
-            publish_time_offset_from_day_start=publish_time_offset_from_day_start,
-        )
+        # Add the publish time
+        df["date"] = df["Interval Start"].dt.date
+        unique_dates = sorted(df["date"].unique())
+
+        # Solar and wind forecast DAM is published at 7:00 AM
+        # http://oasis.caiso.com/mrioasis/logon.do > Atlas Reference > Publications > OASIS Publications Schedule  # noqa
+        for forecast_date in unique_dates:
+            publish_time = (
+                pd.Timestamp(forecast_date, tz=self.default_timezone)
+                # One day because this is a day-ahead forecast
+                - pd.Timedelta(days=1)
+            ).replace(
+                hour=7,
+                minute=0,
+            )
+            df.loc[df["date"] == forecast_date, "Publish Time"] = publish_time
```

Comment:
This is the same logic we now use for the load forecasts as implemented in #522.